crates/bevy_derive/src/derefs.rs

```diff
@@ -19,7 +19,7 @@ pub fn derive_deref(input: TokenStream) -> TokenStream {
     let (impl_generics, ty_generics, where_clause) = ast.generics.split_for_impl();
 
     TokenStream::from(quote! {
-        impl #impl_generics ::std::ops::Deref for #ident #ty_generics #where_clause {
+        impl #impl_generics ::core::ops::Deref for #ident #ty_generics #where_clause {
```

Comment:
Care needs to be taken with these macros, as we _can_ rely on the existence of `core`, but we _can't_ rely on the existence of `alloc`, as the callsite needs to already have `extern crate alloc;` added. Again, this would be addressed by providing a `bevy_alloc` crate, giving a concrete FQN that we can always rely on.